### ISSUE TITLE
Something has changed about true_color

### ISSUE THREAD

**Issue by Julian** [jonathanslenders/ptpython#267]
Hi, apologies for what undoubtedly will be a partial issue report, will update it as I figure out what's up, but in updating recently to a more recent release, something has changed about the way to enable true color support.

[The example](https://github.com/jonathanslenders/ptpython/blob/master/examples/ptpython_config/config.py#L108) still does things the way I do (by setting a `true_color` attribute), but that's the only reference to that attribute in the repo.

The reason I notice is because I am [proactively making sure I don't set useless attributes](https://github.com/Julian/dotfiles/blob/d490b983431db5f4a983af21145d976a5420b1b6/.config/ptpython/config.py#L4-L10) when setting config variables, and that safeguard started now failing and telling me that attribute has disappeared. If that's what's happened (the name of the attribute has changed say), it's IMHO a bad antipattern to have to set attributes on someone else's object like this rather than the object taking them as arguments to `__init__`, for exactly the reason in this ticket -- namely, the attribute disappears, and no one notices.

So, tl;dr, I am like 60% sure that perhaps what's changed is that now the expected API is to set the [`color_depth` attribute](https://github.com/jonathanslenders/ptpython/blob/master/ptpython/python_input.py#L232) to `ColorDepth.24_Bit`, in which case this issue boils down to:

* the config example is out of date
* it uses an "unsafe" way to configure the REPL, so ideally the REPL would grow a safer one

**Comment by jonathanslenders** [comment]
Hi @Julian,
Thanks for reporting this issue! I fixed the example configuration file.

You are absolutely correct that setting attributes like this is not that clean. The reason I have chosen this design is that it allows people to change "any" internal detail of the REPL in the configuration file. It would indeed be cleaner to create some kind of "Configuration" class, and have the configuration file instantiate and return that.

At this point, I'm considering toml configuration files and using tomlkit to change these files whenever the user changes something in the settings menu. This works well for most settings, but I'd still need a Python config file for the people that have custom key bindings.

**Comment by Julian** [comment]
> At this point, I'm considering toml configuration files and using tomlkit to change these files whenever the user changes something in the settings menu.

That'd be great!

And thanks! Think that confirms I was on the right track, closing this I suppose since it sounds like you've got a plan for part B.

Appreciated!
